File: components/netty-http-listener/src/main/kotlin/io/terminus/dalaran/component/netty/http/NettyHttpListener.java
package io.terminus.dalaran.component.netty.http;

import io.terminus.dalaran.BodyMode;
import io.terminus.dalaran.DalaranTrigger;
import io.terminus.dalaran.annotation.DalaranComponent;
import org.apache.camel.model.RouteDefinition;

@DalaranComponent(value = "netty-http-listener", configType = NettyHttpConfig.class, bodyMode = BodyMode.Serialized)
public class NettyHttpListener implements DalaranTrigger<NettyHttpConfig> {

    @Override
    public void buildFromRoute(RouteDefinition route, NettyHttpConfig config) {
        String uri = "netty4-http:" + config.getProtocol().name().toLowerCase() +
                "://" + config.getHost() + ":" + config.getPort() + config.getPath() +
                "?httpMethodRestrict=" + config.getMethod();
        route.from(uri);
    }
}

File: components/netty-http-listener/src/main/kotlin/io/terminus/dalaran/component/netty/http/NettyHttpListener.kt
package io.terminus.dalaran.component.netty.http

import io.terminus.dalaran.DalaranComponent
import io.terminus.dalaran.DalaranTrigger

@DalaranComponent("netty-http-listener", configType = NettyHttpConfig::class)
class NettyHttpListener : DalaranTrigger<NettyHttpConfig> {
    private val camelComponentScheme = "netty4-http"

    override fun buildRouterUri(config: NettyHttpConfig): String =
            "$camelComponentScheme:${config.protocol.value}://${config.host}:${config.port}${config.path}?httpMethodRestrict=${config.method}"
}